Updated troubleshooting section (#16)
ReVanced-Extended-Community/Community-Guides: general-guides/community-wiki/general-troubleshooting.md

> If you patch with the RVX Manager on a different device you'll need to export the patched APK so that you can send it to the device you plan to install it on. To export the APK, tap the save icon when patching is completed as shown in [this image](https://imgur.com/a/FKD0okE). Then press "Save".

If you want to patch on PC, follow the instructions for the desired method in [inotia00's documentation](https://github.com/inotia00/revanced-documentation#revanced-extended-documentation).

### **2. "Feature not implemented: This application is a split APK and cannot be selected."**

On non-root devices you must use a full (non-bundle) APK when patching. Refer to steps 1 and 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/) to download and select the full APK from storage.

### **3. Nothing happens when I select the APK from storage**

Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest). If you already are using the correct version, continue with the following steps:

> **1)** Uninstall the official app you are trying to patch. 

Reviewer: My fault

```suggestion
**1)** Uninstall the unpatched YouTube / YT Music / Reddit from your device. 
```

> If you cannot uninstall it, go to the App info of the app > Options menu (as shown in [this image](https://imgur.com/a/0js3AZR)) > Uninstall updates.

> **2)** Install the APK that you downloaded. (Now you should have the suggested APK installed.)

Reviewer: My fault

```suggestion
**2)** Install the APK that you downloaded from APKMirror.
```

**3)** Open the RVX Manager > Patcher > Select an application, and select the app from the app list.

Continue from the middle of step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/).

### **4. The manager isn't loading patches or applications**

**a)** Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest).

> **b)** Disable any VPNs or ad-blockers and restart the RVX Manager.

Reviewer: ```suggestion
**b)** Disable any ad-blockers or VPNs and restart the RVX Manager.
```

### **5. Patcher is aborting / failing to apply patches**

**a)** Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest).

**b)** Make sure you are using the suggested APK version given in step 1 of the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/).

**c)** If your device's language was set to a language that reads right-to-left when the failure occurred then set your device's language to English or any language that reads left-to-right. Then fully close the Manager, and try again.

> * If it still doesn't work you can try searching and asking on the [subreddit](https://www.reddit.com/r/revancedextended/) or [Telegram group](https://t.me/revanced_extended_chat) for your error. Alternatively, you can patch on a different device or use a different patching method.

Reviewer: ```suggestion
* If it still doesn't work you can try searching and asking for support in the [subreddit](https://www.reddit.com/r/revancedextended/) or [Telegram group](https://t.me/revanced_extended_chat). Alternatively, you can patch on a different device or use a different patching method.
```

* If you want to patch on a different device refer to the info in issue **#1** above.

> ### **6. "App not installed."**
> 
> If you encounter any sort of "App not installed" error during the installation of the patched app, follow these steps to resolve the issue:
> 
> 1. **Google Play Protect Warning:**
>    - If you see a Google Play Protect pop-up, tap **"More details"** and then **"Install anyway"**. Do not tap **"Got it"**, as this will cancel the installation. [View Image](https://imgur.com/a/Ck8nfhn).
> 
> 2. **Ensure Sufficient Storage:**
>     - Make sure your device has enough storage space.
> 
> 3. **Uninstall Existing Versions:**
>    - Remove any existing patched apps, including those in alternate user profiles or private folders.
> 
>  > Youtube/YT Music
> > 
> >  - If you are trying to install the unpatched Youtube/YT Music APK from APKMirror.com, note that it is not necessary to install it directly. Instead, follow step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners) to select and patch the APK correctly.
> 
> 
> 4. **Check APK Compatibility:**
>    - Ensure that the patched APK matches the architecture of your device. Refer to step 1 in the [general guide](https://github.com/ReVanced-Extended-Community/Community-Guides/blob/main/general-guides/community-wiki/ytm-guide.md#1-downloading-rvx-manager-yt-music-apk--vanced-microg) for information on downloading the correct APK.
> 
> > Youtube/YT Music
> >  - If you are trying to install the unpatched Youtube/YT Music APK from APKMirror.com, note that it is not necessary to install it directly. Instead, follow step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners) to select and patch the APK correctly.
> 
> **Youtube/YT Music specific issues**
> 
> 5. **Patch with `GmsCore Support`:**
>    - Ensure that the `GmsCore Support` patch is applied when patching the APK.
> 
> 6. **GmsCore Version:**
>    - Ensure you are not installing an older version of GmsCore than the one already installed. Uninstall the current version via Settings > Apps > GmsCore > Uninstall.
>    - Make sure to download the latest GmsCore from [here](https://github.com/ReVanced/GmsCore/releases/latest)
> 
> By following these steps, you should be able to resolve the majority of installation issues with Youtube/YT Music Extended.

Reviewer: ```suggestion
### **6. Installation errors**

If you encounter any installation error when installing the patched app, follow these steps until the issue is resolved:

1. **Google Play Protect Warning:**
If you see a Google Play Protect pop-up, tap **"More details"** and then **"Install anyway"**. Do not tap "Got it", as this will cancel the installation. [View Image](https://imgur.com/a/Ck8nfhn).

2. **Ensure Sufficient Storage:**
Make sure your device has enough storage space.

3. **Uninstall Existing Versions:**
Remove any existing installation of the app you are trying to install, even if it is in an alternate user profile or private folder.

4. **Check The APK**
If you are trying to install the unpatched APK from APKMirror.com, note that it is simpler to avoid installing it before patching. Instead, follow step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners) to select and patch the APK without installing it beforehand.

Youtube / YT Music:
> 5. **Patch with `GmsCore Support`:**
> Ensure that the `GmsCore Support` patch is applied when patching the APK.

YT Music:
> 6. **Check APK Compatibility:**
> Ensure that the APK you patched is built for your device's architecture. Refer to TODO for information on downloading the correct APK.

```

> ### **7. App crashes on startup**
> 
> If the app is crashing on startup, there are 3 common causes of it:
> 
> YouTube and YT Music:
> > **a)** You do not have GmsCore installed. Download and install [GmsCore](https://github.com/ReVanced/GmsCore/releases/latest).
> 
> **b)** There was a failure or error while it was patching and you installed the broken APK. Refer to troubleshooting issue **#6** if this occurred.
> 

Reviewer: ```suggestion
### **7. App crashes on startup**

If the app is crashing on startup, it may be because of one of the following:

1. There was a failure or error while it was patching and you installed the broken APK. Refer to troubleshooting issue **#5** if this occurred.

YouTube / YT Music:
> 2. You do not have GmsCore installed. Download and install [GmsCore](https://github.com/ReVanced/GmsCore/releases/latest).
>
> 3. The `GmsCore support` patch was not applied to the app.

YT Music:
> 4. The email you are signing in with has parental control restrictions.

```

# **General Troubleshooting**

### **1. My device is not supported**

You'll need to patch on a different Android device or on PC.

> When patching on a different device or PC, be sure to patch the APK variant that was built for the architecture of the device you plan to install the patched APK on. For YouTube, you can simply use the universal version, but for YouTube Music, you'll need the specific version corresponding to your device.

Reviewer: ```suggestion
YT Music:
> Note that when patching YT Music on a different device or PC, you'll need to patch the APK variant that was built for the architecture of the device you plan to **install** the patched APK on. 
```

I haven't yet added the info for selecting the right APK to the setup info guide, but after I complete it this can link to it.

If you want to patch on PC, follow the instructions for the desired method in [inotia00's documentation](https://github.com/inotia00/revanced-documentation#revanced-extended-documentation).

### **2. "Feature not implemented: This application is a split APK and cannot be selected."**

On non-root devices you must use a full (non-bundle) APK when patching. Refer to steps 1 and 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/) to download and select the full APK from storage.

### **3. Nothing happens when I select the Youtube/YT Music APK from storage**

Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest). If you already are using the correct version, continue with the following steps:

**1)** Go to the App info of stock YouTube/YT Music > Options menu (as shown in [this image](https://imgur.com/a/0js3AZR)) > Uninstall updates (or fully uninstall if you can).

**2)** Install the YouTube/YT Music APK that you downloaded

**3)** Open the RVX Manager > Patcher > Select an application > **YouTube/YT Music**.

> Continue from the middle of step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/).

Reviewer: This entire section should ideally also be written to include Reddit, but it makes it complicated (since the Reddit package name isn't changed when it is patched).

LMK what you think about this:

```suggestion
### **3. Nothing happens when I select the APK from storage**

Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest). If you already are using the correct version, continue with the following steps:

**1)** Uninstall the official app you are trying to patch. 

> If you cannot uninstall it, go to the App info of the app > Options menu (as shown in [this image](https://imgur.com/a/0js3AZR)) > Uninstall updates.

**2)** Install the APK that you downloaded. (Now you should have the suggested APK installed.)

**3)** Open the RVX Manager > Patcher > Select an application, and select the app from the app list.

Continue from the middle of step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/).
```

### **4. The manager isn't loading patches or applications**

**a)** Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest).

**b)** Disable any VPNs or ad-blockers and restart the RVX Manager.

### **5. Patcher is aborting / failing to apply patches**

**a)** Make sure you are using the latest [RVX Manager](https://github.com/inotia00/revanced-manager/releases/latest).

> **b)** Make sure you are using the suggested Youtube/YT Music version given in step 1 of the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/).

Reviewer: ```suggestion
**b)** Make sure you are using the suggested APK version given in step 1 of the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners/).
```

* If it still doesn't work you can try searching and asking on the [subreddit](https://www.reddit.com/r/revancedextended/) or [Telegram group](https://t.me/revanced_extended_chat) for your error. Alternatively, you can patch on a different device or use a different patching method.

* If you patch on a different device or use a different patching method be sure to patch the APK variant that was built for the architecture of the device you plan to **install** the patched APK on. Refer to step 1 in the [general guide](https://github.com/ReVanced-Extended-Community/Community-Guides/blob/main/general-guides/community-wiki/ytm-guide.md#1-downloading-rvx-manager-yt-music-apk--vanced-microg) for info on downloading an APK for a specific architecture.

> If you patch with the RVX Manager on a different device you'll need to export the patched APK so that you can send it to the device you plan to install it on. To export the APK, tap the save icon when patching is completed as shown in [this image](https://imgur.com/a/FKD0okE). Then press "Save".

> If you want to patch on PC, follow the instructions for the desired method in in [inotia00's documentation](https://github.com/inotia00/revanced-documentation#revanced-extended-documentation).

Reviewer: ```suggestion
* If you want to patch on a different device refer to the info in issue **#1** above.
```

### **6. Manager is stuck on "Installing..."**

> Export the patched APK by tapping the save icon as shown in [this image](https://imgur.com/a/FKD0okE). Save it, and then install the exported APK from your file manager. You might get an error while installing it, if so refer to the entry below for instructions on how to solve the issue.

Reviewer: This bug doesn't occur anymore

```suggestion

```

> ### **7. "App not installed."**

Reviewer: This issue should ideally also have Reddit in mind.

For parts of the checklist specific to YT/YT Music, you can put it as a quoted note thing (idk the term lol). Look at section 2b in the setup info guide to see how I divided the info specific to yt and ytm

https://github.com/ReVanced-Extended-Community/Community-Guides/blob/wiki-redesign/general-guides/community-wiki/patching%20%26%20setup%20info.md#2b-downloader-setup

If you encounter any sort of "App not installed" error during the installation of Youtube/YT Music Extended, follow these steps to resolve the issue:

1. **Google Play Protect Warning:**
   - If you see a Google Play Protect pop-up, tap **"More details"** and then **"Install anyway"**. Do not tap **"Got it"**, as this will cancel the installation. [View Image](https://imgur.com/a/Ck8nfhn).

2. **Ensure Sufficient Storage:**

>     - Make sure your device has enough storage space

Reviewer: ```suggestion
2. **Ensure Sufficient Storage:**
    - Make sure your device has enough storage space.
```

   - Ensure that the `GmsCore Support` patch is applied when patching the APK.

5. **Check APK Compatibility:**
   - If you are trying to install the unpatched Youtube/YT Music APK from APKMirror.com, note that it is not necessary to install it directly. Instead, follow step 2 in the [beginners guide](https://www.reddit.com/r/revancedextended/comments/12vxggr/revanced_extended_guide_for_beginners) to select and patch the APK correctly.
   - Ensure that the patched APK matches the architecture of your device. Refer to step 1 in the [general guide](https://github.com/ReVanced-Extended-Community/Community-Guides/blob/main/general-guides/community-wiki/ytm-guide.md#1-downloading-rvx-manager-yt-music-apk--vanced-microg) for information on downloading the correct APK.

6. **GmsCore Version:**
   - Ensure you are not installing an older version of GmsCore than the one already installed. Uninstall the current version via Settings > Apps > GmsCore > Uninstall.

By following these steps, you should be able to resolve the majority of installation issues with Youtube/YT Music Extended.

### **8. App crashes on startup**

If the app is crashing on startup, there are 3 common causes of it:

> **a)** You do not have GmsCore installed. Download and install [GmsCore](https://github.com/ReVanced/GmsCore/releases/latest).

Reviewer: Ofc this is specific to YT/YT Music

```suggestion
YouTube and YT Music:
> **a)** You do not have GmsCore installed. Download and install [GmsCore](https://github.com/ReVanced/GmsCore/releases/latest).
```

**b)** There was a failure or error while it was patching and you installed the broken APK. Refer to troubleshooting issue **#6** if this occurred.

> **c)** The email you are signing in with has parental control restrictions.

Reviewer: The email one is specific to YTM.

Author: Done
